jax/experimental/jax2tf/jax2tf.py

```diff
@@ -1373,38 +1375,46 @@ def wrapper(*args: TfVal, **kwargs):
     else:
       # All argnums should be boolean
       assert len(argnum_types) == 1, argnum_types
-      args_cast = [(tf.cast(a, tf.int8) if i in argnums else a)
-                   for i, a in enumerate(args)]
-      if "_in_avals" in kwargs:
-
-        def cast_aval(aval):
-          assert aval.dtype == np.bool_
-          return core.ShapedArray(aval.shape, np.int8)
-
-        _in_avals_cast = [
-            cast_aval(aval) if i in argnums else aval
-            for i, aval in enumerate(kwargs["_in_avals"])
-        ]
-        _out_aval_cast = tf.nest.map_structure(cast_aval, kwargs["_out_aval"])
-        kwargs = dict(
-            kwargs, _in_avals=_in_avals_cast, _out_aval=_out_aval_cast)
-      out = f(*args_cast, **kwargs)
-      return tf.nest.map_structure(lambda o: tf.cast(o, tf.bool), out)
+      if boolean_f != None:
+        return boolean_f(*args, **kwargs)
+      else:
+        args_cast = [(tf.cast(a, tf.int8) if i in argnums else a)
+                    for i, a in enumerate(args)]
+        if "_in_avals" in kwargs:
+
+          def cast_aval(aval):
+            assert aval.dtype == np.bool_
+            return core.ShapedArray(aval.shape, np.int8)
+
+          _in_avals_cast = [
+              cast_aval(aval) if i in argnums else aval
+              for i, aval in enumerate(kwargs["_in_avals"])
+          ]
+          _out_aval_cast = tf.nest.map_structure(cast_aval, kwargs["_out_aval"])
+          kwargs = dict(
+              kwargs, _in_avals=_in_avals_cast, _out_aval=_out_aval_cast)
+        out = f(*args_cast, **kwargs)
+        return tf.nest.map_structure(lambda o: tf.cast(o, tf.bool), out)
 
   return wrapper
 
 
-tf_impl[lax.or_p] = bool_to_int8(tf.bitwise.bitwise_or, argnums=(0, 1))
-tf_impl[lax.and_p] = bool_to_int8(tf.bitwise.bitwise_and, argnums=(0, 1))
-tf_impl[lax.xor_p] = bool_to_int8(tf.bitwise.bitwise_xor, argnums=(0, 1))
+tf_impl[lax.or_p] = handle_boolean_args(tf.bitwise.bitwise_or, argnums=(0, 1), boolean_f=tf.logical_or)
+tf_impl[lax.and_p] = handle_boolean_args(tf.bitwise.bitwise_and, argnums=(0, 1), boolean_f=tf.logical_and)
+tf_impl[lax.xor_p] = handle_boolean_args(tf.bitwise.bitwise_xor, argnums=(0, 1), boolean_f=tf.math.logical_xor)
 
 tf_impl[lax.eq_p] = tf.math.equal
 tf_impl[lax.ne_p] = tf.math.not_equal
 
-tf_impl[lax.ge_p] = bool_to_int8(tf.math.greater_equal, argnums=(0, 1))
-tf_impl[lax.gt_p] = bool_to_int8(tf.math.greater, argnums=(0, 1))
-tf_impl[lax.le_p] = bool_to_int8(tf.math.less_equal, argnums=(0, 1))
-tf_impl[lax.lt_p] = bool_to_int8(tf.math.less, argnums=(0, 1))
+bool_greater = lambda x,y: tf.logical_and(x, tf.logical_not(y)) # Only one combo: T,F -> T
+bool_less = lambda x,y: tf.logical_and(tf.logical_not(x), y) # Only one combo: F,T -> T
+def accept_equal(fn):
```

Comment:
I would make the name of this function more specific, e.g., `boolean_compare_or_equal`.

Reply:
It can also be greater_than_or_equal is not(less_than). This is shorter code, but possibly more confusing?

Reply:
I have split into two functions and used the >= is !(<) trick to simplify it